mypy/semanal.py

```diff
@@ -1770,11 +1770,12 @@ def check_and_set_up_type_alias(self, s: AssignmentStmt) -> None:
         lvalue = s.lvalues[0]
         if not isinstance(lvalue, NameExpr):
             return
-        if (len(s.lvalues) == 1 and not self.is_func_scope() and
-                not (self.type and isinstance(s.rvalue, NameExpr) and lvalue.is_def)
+        if (len(s.lvalues) == 1 and
+                not ((self.type or self.is_func_scope()) and
```

Comment:
This condition is hard to understand. Add a comment describing what it does at a high level and maybe also give some concrete examples. It's also worth trying to simplify the nested boolean expression. Ideas:

* Make the top-level expression of form `e1 and (...) and not e2` instead of `e1 and not (...) and not e2` -- the first `not` operator makes this a bit harder to understand, I think.
* Use a temp variable to hold the result of `self.type or self.is_func_scope()` and give it a semantically meaningful name.